src/App.tsx

```diff
@@ -1,10 +1,84 @@
-import React from 'react';
+import React, { FC } from 'react';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+import { TodoList } from './components/TodoList';
 import './App.css';
+import { getTodos, getUsers } from './api/getData';
+import { Actions } from './components/Actions';
+import * as actions from './redux/actions';
 
-const App = () => (
-  <div className="App">
-    <h1>Redux list of todos</h1>
-  </div>
-);
+interface Methods {
+  setIsLoaded: (status: boolean) => void;
+  setIsLoading: (status: boolean) => void;
+  setTodos: (todos: Todo[]) => void;
+}
 
-export default App;
+type Props = Methods & {
+  isLoaded: boolean;
+  isLoading: boolean;
+};
+
+
+const AppTemplate: FC<Props> = ({
+  isLoaded, isLoading, setIsLoaded,
+  setIsLoading, setTodos,
+}) => {
+  async function handleStart() {
+    setIsLoading(true);
+
+    const [
+      todosFromServer,
+      usersFromServer,
+    ] = await Promise.all([
+      getTodos(),
+      getUsers(),
+    ]).finally(() => setIsLoaded(true));
+
+    setTodos(todosFromServer.map((todo) => (
+      {
+        ...todo,
+        user: usersFromServer
+          .find((user) => user.id === todo.userId),
+      })));
+
+    setIsLoading(false);
+  }
+
+  return (
+    <>
+      {isLoaded ? (
+        <>
+          <Actions />
+          <TodoList />
+        </>
+      ) : (
+        <button
+          className="button"
+          type="button"
+          onClick={handleStart}
+        >
+             Press to start
+        </button>
+      )}
+
+      {isLoading && <p>Loading...</p>}
+    </>
+  );
+};
+
+const mapStateToProps = (state: { loadReducer: LoadState }) => {
+  return {
+    isLoaded: state.loadReducer.isLoaded,
+    isLoading: state.loadReducer.isLoading,
+  };
+};
+
+const mapDispatchToProps = (dispatch: Dispatch) => {
+  return {
+    setIsLoaded: (status: boolean) => dispatch(actions.setIsLoaded(status)),
+    setIsLoading: (status: boolean) => dispatch(actions.setIsLoading(status)),
+    setTodos: (todos: Todo[]) => dispatch(actions.setTodos(todos)),
+  };
+};
```

Comment:
make it easy with object
<img width="946" alt="Screenshot 2020-03-16 at 07 30 06" src="https://user-images.githubusercontent.com/36257947/76725856-2e500300-6758-11ea-9230-75cd27488bb9.png">